src/mock/genesis_builder.rs

```diff
@@ -56,15 +56,10 @@ impl GenesisBuilder {
         if !self.spentbook_nodes.is_empty() {
             // we only support a single mock spentbook section.  pubkeys must match.
             assert_eq!(
-                spentbook_node
-                    .key_manager
-                    .public_key_set()
-                    .unwrap()
-                    .public_key(),
+                spentbook_node.key_manager.public_key_set().public_key(),
                 self.spentbook_nodes[0]
                     .key_manager
                     .public_key_set()
-                    .unwrap()
                     .public_key()
```

Comment:
You should be able to compare the `public_key_set()`'s themselves, shouldn't need to convert to `public_key()`